src/controllers/plugins/Plugin.hpp

```diff
@@ -85,10 +87,50 @@ class Plugin
         return this->loadDirectory_;
     }
 
+    // Note: The CallbackFunction object's destructor will remove the function from the lua stack
+    std::optional<lua::CallbackFunction<lua::api::CompletionList, QString,
+                                        QString, int, bool>>
+        getCompletionCallback()
+    {
```

Comment:
Move this function to the source file, and let's maybe add a `using LuaCompletionCallback = ...`; to make this look a bit more sane